packages/frontend-api/src/Model/Resolver/Category/CategoryResolver.php

```diff
@@ -18,35 +22,157 @@ class CategoryResolver implements ResolverInterface, AliasedInterface
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+        $this->domain = $domain;
+    }
+
+    /**
+     * @required
+     * @param \Shopsys\FrontendApiBundle\Model\FriendlyUrl\FriendlyUrlFacade $friendlyUrlFacade
+     * @internal This function will be replaced by constructor injection in next major
+     */
+    public function setFriendlyUrlFacade(FriendlyUrlFacade $friendlyUrlFacade): void
+    {
+        if ($this->friendlyUrlFacade !== null && $this->friendlyUrlFacade !== $friendlyUrlFacade) {
+            throw new BadMethodCallException(sprintf(
+                'Method "%s" has been already called and cannot be called multiple times.',
+                __METHOD__
+            ));
+        }
+        if ($this->friendlyUrlFacade !== null) {
+            return;
+        }
+        @trigger_error(
+            sprintf(
+                'The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+        $this->friendlyUrlFacade = $friendlyUrlFacade;
     }
 
     /**
      * @param string $uuid
      * @return \Shopsys\FrameworkBundle\Model\Category\Category
+     * @deprecated This method will be removed in next major version. It has been replaced by resolveByUuidOrUrlSlug
      */
     public function resolver(string $uuid): Category
     {
+        @trigger_error(
+            sprintf(
+                'The %s() method is deprecated and will be removed in the next major. Use resolveByUuidOrUrlSlug() instead.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
         try {
             return $this->categoryFacade->getByUuid($uuid);
         } catch (CategoryNotFoundException $categoryNotFoundException) {
             throw new UserError($categoryNotFoundException->getMessage());
         }
     }
 
+    /**
+     * @param string|null $uuid
+     * @param string|null $urlSlug
+     * @return \Shopsys\FrameworkBundle\Model\Category\Category
+     */
+    public function resolveByUuidOrUrlSlug(?string $uuid = null, ?string $urlSlug = null): Category
+    {
+        try {
+            if ($uuid !== null) {
+                return $this->getByUuid($uuid);
+            }
+
+            if ($urlSlug !== null) {
+                return $this->getVisibleOnDomainAndSlug($urlSlug);
+            }
+        } catch (FriendlyUrlNotFoundException | CategoryNotFoundException $categoryNotFoundException) {
+            throw new UserError($categoryNotFoundException->getMessage());
+        }
+
+        throw new UserError('You need to provide argument \'uuid\' or \'urlSlug\'.');
+    }
+
     /**
      * @return string[]
      */
     public static function getAliases(): array
     {
         return [
-            'resolver' => 'category',
+            'resolveByUuidOrUrlSlug' => 'resolveByUuidOrUrlSlug',
```

Comment:
I think such a common name (`resolveByUuidOrUrlSlug`) as a resolver alias can lead to confusion.

I suggest something like 
`'resolveByUuidOrUrlSlug' => 'categoryByUuidOrUrlSlug'`

And is a reason to remove the old alias (`category`)? I know it's not used, but in theory, if someone already uses frontend API and override definition, this will be a breaking change for them.
I believe deprecate the method, but keep the resolver alias available is a good compromise, what do you think?

Reply:
:+1: fixed